firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDataMigrationManager.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import com.google.firebase.firestore.auth.User;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.mutation.MutationBatch;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/** Manages SQLite data migration required by SDK version upgrades. */
+public class SQLiteDataMigrationManager implements DataMigrationManager {
+  private final SQLitePersistence db;
+  private final Set<SQLitePersistence.DataMigration> migrations;
+
+  /**
+   * Creates a new data migration manager.
+   *
+   * @param persistence The underlying SQLite Persistence to use for data migrations.
+   * @param migrations The set of migrates needed to run.
+   */
+  public SQLiteDataMigrationManager(
+      SQLitePersistence persistence, Set<SQLitePersistence.DataMigration> migrations) {
+    this.db = persistence;
+    this.migrations = migrations;
+  }
+
+  @Override
+  public void run() {
+    for (SQLitePersistence.DataMigration migration : migrations) {
+      switch (migration) {
+        case BuildOverlays:
```

Comment:
Do we need the migration type? Could me just submit "Runnables" to this class?  Each migration could then simply be a runable job that we execute.

Reply:
I am going to punt it due to laziness. Let's also see what the next data migration looks like first :-)